x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/analysis/analyzer/Analyzer.java

```diff
@@ -341,6 +343,22 @@ else if (plan instanceof Aggregate) {
                     List<Expression> groupings = a.groupings();
                     List<Expression> newGroupings = new ArrayList<>();
                     AttributeMap<Expression> resolved = Expressions.aliases(a.aggregates());
+
+                    //if all we're selecting are functions or literals, even if we're giving them aliases.
+                    final boolean[] updateResolved = {true};
+                    a.forEachUp(p -> p.forEachExpressionsUp(e -> {
```

Comment:
I think here you want to check only the `a.aggregates()` and not all the expressions (including the grouping).
So probably you need the following :
```
a.aggregates().forEach(e -> e.forEachUp(f -> {
    if (f instanceof Function == false && f.foldable() == false) {
        updateResolved.set(Boolean.FALSE);
    }
}));
```